### PR TITLE
Handle no thumbprint/no certificate variable for a binding

### PR DESCRIPTION
**Pull request by matt-richardson**
Relates to https://github.com/OctopusDeploy/OctopusDeploy/pull/3705 and https://github.com/OctopusDeploy/Issues/issues/5480

### CODE REVIEW

File: source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1 [hunk 1]
```diff
@@ -467,12 +467,15 @@ if ($deployAsWebSite)
 		# in the deployment process
 		if ($_.certificateVariable) {
 			$sslCertificateThumbprint = $OctopusParameters[$_.certificateVariable + ".Thumbprint"]
-		} else {
+		} elseif ($_.thumbprint){
 			# Otherwise, the certificate thumbprint was supplied directly in the binding
 			$sslCertificateThumbprint = $_.thumbprint.Trim()
+		} else {
+			[Console]::Error.WriteLine("Cannot configure a https binding when neither certificate thumbprint nor certificate variable is not supplied on the binding.")
```
Comment (michaelnoonan) [inline source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1]: I got confused for the first 3 attempts at reading this. :)

```suggestion
			Write-Error "To configure an HTTPS binding please choose a certificate which is managed by Octopus, or provide the thumbprint for a certificate which is already available in the Machine certificate store."
```

File: source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1 [hunk 2]
```diff
@@ -467,12 +467,15 @@ if ($deployAsWebSite)
...
+			exit 1
 		}
 
-		Write-Host "Finding SSL certificate with thumbprint $sslCertificateThumbprint"
+		Write-Host "Finding SSL certificate with thumbprint $sslC\ertificateThumbprint"
```
Comment (michaelnoonan) [inline source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1]: Is the `\` PowerShell magic? `sslC\ertificateThumbprint`.

File: source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1 (hunk 1 above)
Comment (matt-richardson) [reply source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1]: Yeah - i had trouble coming up with the words.
I was using `Error.WriteLine(...)` as the `Write-Error` was being truncated and wasn't readable.

File: source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1 (hunk 2 above)
Comment (matt-richardson) [reply source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1]: PEBKAC issue. Fixed.

**Pull request merged**